hal/ia32/_interrupts.S

```diff
@@ -24,17 +24,27 @@
 
 
 #ifndef NDEBUG
-#define CTXPUSHL 14
+#define CTXPUSHL 42 /* 10 + 4 (DEBUG) + 28 (FPU) */
 #else
-#define CTXPUSHL 10
+#define CTXPUSHL 38 /* 10 + 28 (FPU) */
 #endif
 
+#define EAX_OFFSET 124 //(16 + FPU_CONTEXT_SIZE)
+
 
 .global interrupts_pushContext
 interrupts_pushContext:
 	xchgl (%esp), %edx
 	movl %edx, -(4 * CTXPUSHL)(%esp)
 	popl %edx
+	/* Check TS flag in CR0 register */
+	pushl %eax
+	movl %cr0, %eax
+	andl $CR0_TS_BIT, %eax
+	xchgl (%esp), %eax
+	jz .interrupts_pushFPUContext
+	subl $FPU_CONTEXT_SIZE, %esp
```

Comment:
I suggest `jnz` to `interrupts_pushRegisters` to skip FPU context part instead of jumping back and forth, we subtract from esp in both cases